src/Microsoft.VisualStudio.Extensibility.Testing.Xunit.IntegrationTests.Shared/VisualStudioInstanceKeyTests.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for more information.
+
+namespace Microsoft.VisualStudio.Extensibility.Testing.Xunit.IntegrationTests
+{
+    using System;
+    using global::Xunit;
+    using global::Xunit.Harness;
+
+    public class VisualStudioInstanceKeyTests
+    {
+        private static readonly Random _random = new Random(Environment.TickCount);
+
+        private static int NextInt(int maxValue)
+        {
+            lock (_random)
+            {
+                return _random.Next(maxValue);
+            }
+        }
+
+        [Theory]
+        [InlineData(new object[] { new string[] { } })]
+        [InlineData(new object[] { new string[] { "Key=Value" } })]
+        [InlineData(new object[] { new string[] { "Key=" } })]
+        [InlineData(new object[] { new string[] { "Key==", "Key=;" } })]
+        [InlineData(new object[] { new string[] { "DOTNET_MULTILEVEL_LOOKUP=", "DOTNET_INSTALL_DIR=", "DotNetRoot=", "DotNetTool=" } })]
+        public void TestSerializationWithEnvironmentVariables(string[] environmentVariables)
+        {
+            var versions = (VisualStudioVersion[])Enum.GetValues(typeof(VisualStudioVersion));
+            var version = versions[NextInt(versions.Length)];
+
+            var rootSuffix = NextInt(2) switch
```

Comment:
I don't quite understand why this is random - I can understand not wanting to test the entire matrix of version + exp, but would all versions + one or two on exp be reasonable?  

Just wary of a test that might randomly fail depending on the version combination it selects

Reply:
💭 In the event of failure, it should be easy enough to use the new Run Until Failure feature of Test Explorer to find a failing case.